### PR TITLE
Assessor can now view and add communication log entries

### PR DESCRIPTION
**Pull request by tony-gaia**
Added comm log functionality to assessor views and templates, namely the
forms, panels and modals.

Show/Add comm log now uses OfficerOrAssessorRequired mixin rather than
OfficerRequired

Changed 'officer' on Comm Log modal to 'staff' as it can now also be an
assessor.

Renamed assessor_conditions_read_only html/js to be
view_readonly_assessor to be consistent with officer versions of these.

Added a util function to get the default comm log "to" from an
application and refactored the various application views to use this
function.

Fixed bug for sending an email with no attachments.

### CODE REVIEW

File: wildlifelicensing/apps/emails/emails.py [hunk 1]
```diff
@@ -69,6 +69,9 @@ def send(self, to_addresses, from_address=None, context=None, attachments=None,
             to_addresses = [to_addresses]
         if attachments is not None and not isinstance(attachments, list):
             attachments = list(attachments)
+        else:
+            attachments = []
+
```
Comment (serge-gaia) [inline wildlifelicensing/apps/emails/emails.py:75]: I just realized that this is wrong. 
The else should be replaced by:

```
if attachments is None:
    attachments = []
```

Currently if we pass a `list` of attachments it get converted to `[]`

File: wildlifelicensing/apps/main/models.py [hunk 2]
```diff
@@ -122,7 +122,7 @@ class CommunicationsLogEntry(models.Model):
     documents = models.ManyToManyField(Document, blank=True)
 
     customer = models.ForeignKey(EmailUser, null=True, related_name='customer')
-    officer = models.ForeignKey(EmailUser, null=True, related_name='officer')
+    staff = models.ForeignKey(EmailUser, null=True, related_name='officer')
```
Comment (serge-gaia) [inline wildlifelicensing/apps/main/models.py:126]: Should change also the `related_name`

File: wildlifelicensing/apps/emails/emails.py (hunk 1 above)
Comment (tony-gaia) [reply wildlifelicensing/apps/emails/emails.py:75]: Changed as suggested.

File: wildlifelicensing/apps/main/models.py (hunk 2 above)
Comment (tony-gaia) [reply wildlifelicensing/apps/main/models.py:126]: Changed and checked if anything was using the officer related name, and it wasn't.

**Pull request merged**